### PR TITLE
chore: Table Loading Improvements

### PR DESCRIPTION
**Pull request by jadebenn**
Make more use of emplace() in some of the CDClient tables

### CODE REVIEW

File: dDatabase/CDClientDatabase/CDClientTables/CDLootTableTable.cpp [hunk 1]
```diff
@@ -79,10 +79,10 @@ const LootTableEntries& CDLootTableTable::GetTable(uint32_t tableId) {
 
 	while (!tableData.eof()) {
 		CDLootTable entry;
-		entries[tableId].push_back(ReadRow(tableData));
+		entries.at(tableId).emplace_back(ReadRow(tableData));
```
Comment (EmosewaMC) [inline dDatabase/CDClientDatabase/CDClientTables/CDLootTableTable.cpp:82]: This is already a container with safe subscripting.  At is not necessary here and should be reverted.

Comment (jadebenn) [reply dDatabase/CDClientDatabase/CDClientTables/CDLootTableTable.cpp:82]: Are we intentionally keeping the possibility of it creating a new entry here?

Comment (EmosewaMC) [reply dDatabase/CDClientDatabase/CDClientTables/CDLootTableTable.cpp:82]: yes, that is the whole point of this and why i wrote it like that.

File: dDatabase/CDClientDatabase/CDClientTables/CDMissionTasksTable.h [hunk 2]
```diff
@@ -25,7 +27,7 @@ class CDMissionTasksTable : public CDTable<CDMissionTasksTable, std::vector<CDMi
 	// Queries the table with a custom "where" clause
 	std::vector<CDMissionTasks> Query(std::function<bool(CDMissionTasks)> predicate);
 
-	std::vector<CDMissionTasks*> GetByMissionID(uint32_t missionID);
+	std::vector<CDMissionTasks*> GetByMissionID(const uint32_t missionID);
```
Comment (EmosewaMC) [inline dDatabase/CDClientDatabase/CDClientTables/CDMissionTasksTable.h:30]: The corresponding function in the cpp file was left unchanged and no compiler error was detected.  This function can be removed and re-added if necessary.

File: dDatabase/CDClientDatabase/CDClientTables/CDObjectsTable.cpp [hunk 3]
```diff
@@ -1,7 +1,7 @@
 #include "CDObjectsTable.h"
 
 namespace {
-	CDObjects m_default;
+	CDObjects objDefault;
```
Comment (EmosewaMC) [inline dDatabase/CDClientDatabase/CDClientTables/CDObjectsTable.cpp:4]: file scope should be PascalCase.  Helps to discern what is a global and what isnt.  only reason this wasnt is because I was editing too many files at that time to warrant variable name changes.

File: dGame/dComponents/PetComponent.cpp [hunk 4]
```diff
@@ -70,8 +70,8 @@ std::map<LOT, int32_t> PetComponent::petFlags = {
 		{ 13067, 838 }, // Skeleton dragon
 };
 
-PetComponent::PetComponent(Entity* parentEntity, uint32_t componentId) : Component{ parentEntity } {
-	m_PetInfo = CDClientManager::GetTable<CDPetComponentTable>()->GetByID(componentId); // TODO: Make reference when safe
+PetComponent::PetComponent(Entity* parentEntity, uint32_t componentId) : Component{ parentEntity },
+	m_PetInfo{ CDClientManager::GetTable<CDPetComponentTable>()->GetByID(componentId) } {
```
Comment (EmosewaMC) [inline dGame/dComponents/PetComponent.cpp:74]: Please pull out to another PR.  On this note, the object in question was not updated to be a reference which should be done as well.

File: dDatabase/CDClientDatabase/CDClientTables/CDPackageComponentTable.cpp [hunk 5]
```diff
@@ -19,12 +19,11 @@ void CDPackageComponentTable::LoadValuesFromDatabase() {
 	// Now get the data
 	auto tableData = CDClientDatabase::ExecuteQuery("SELECT * FROM PackageComponent");
 	while (!tableData.eof()) {
-		CDPackageComponent entry;
+		auto &entry = entries.emplace_back();
```
Comment (EmosewaMC) [inline dDatabase/CDClientDatabase/CDClientTables/CDPackageComponentTable.cpp:22]: format: we use left aligned ref and ptr.

File: dDatabase/CDClientDatabase/CDClientTables/CDPhysicsComponentTable.cpp [hunk 6]
```diff
@@ -4,32 +4,31 @@ void CDPhysicsComponentTable::LoadValuesFromDatabase() {
...
+		UNUSED_ENTRY(entry.rotSpeed = tableData.getFloatField("rotSpeed", -1),)
+		entry.playerHeight = static_cast<float>(tableData.getFloatField("playerHeight")),
+		entry.playerRadius = static_cast<float>(tableData.getFloatField("playerRadius")),
+		entry.pcShapeType = tableData.getIntField("pcShapeType"),
+		entry.collisionGroup = tableData.getIntField("collisionGroup"),
+		UNUSED_ENTRY(entry.airSpeed = tableData.getFloatField("airSpeed"),)
+		UNUSED_ENTRY(entry.boundaryAsset = tableData.getStringField("boundaryAsset"),)
+		UNUSED_ENTRY(entry.jumpAirSpeed = tableData.getFloatField("jumpAirSpeed"),)
+		UNUSED_ENTRY(entry.friction = tableData.getFloatField("friction"),)
+		UNUSED_ENTRY(entry.gravityVolumeAsset = tableData.getStringField("gravityVolumeAsset"),)
```
Comment (EmosewaMC) [inline dDatabase/CDClientDatabase/CDClientTables/CDPhysicsComponentTable.cpp:25]: why was every semi colon replaced with a comma?

**Review by EmosewaMC — CHANGES_REQUESTED** [review]
good improvements to construction of objects in containers.  As you know unnecessary copies are good to remove.  The pet component changes can be moved to their own pr (small enough to anyways and arent really related to any code changes here) and the .at revert is necessary as the table was complete before as far as functionality goes, and this change will cause crashes when they are not necessary.

File: dDatabase/CDClientDatabase/CDClientTables/CDPhysicsComponentTable.cpp (hunk 6 above)
Comment (jadebenn) [reply dDatabase/CDClientDatabase/CDClientTables/CDPhysicsComponentTable.cpp:25]: Genuinely unsure what happened there

File: dDatabase/CDClientDatabase/CDClientTables/CDPackageComponentTable.cpp (hunk 5 above)
Comment (jadebenn) [reply dDatabase/CDClientDatabase/CDClientTables/CDPackageComponentTable.cpp:22]: My spacebar evidently is a C purist 😅

File: dGame/dComponents/PetComponent.cpp (hunk 4 above)
Comment (jadebenn) [reply dGame/dComponents/PetComponent.cpp:74]: I no longer want to include a reference member due to it inhibiting move constructors and reducing data locality.

File: dDatabase/CDClientDatabase/CDClientTables/CDObjectsTable.cpp (hunk 3 above)
Comment (jadebenn) [reply dDatabase/CDClientDatabase/CDClientTables/CDObjectsTable.cpp:4]: Not sure I agree with that formatting decision. Isn't it much more important to preserve the clear distinction between functions and objects than to convey the file scope? One is a very easy to catch compiler error (if not quickly found by the red squiggles in an IDE). The other could introduce very subtle and hard to diagnose bugs if a PascalCase variable was somehow used in a context where a function or object was expected instead.

In particular, I really don't like how your proposed formatting rule would interact with constructor calls. ObjName objName{}; is much clearer than ObjName ObjName{}; If you use parenthesis, the latter might even look like a function call: ObjName ObjName().

Comment (EmosewaMC) [reply dDatabase/CDClientDatabase/CDClientTables/CDObjectsTable.cpp:4]: Resolution agreed upon is f_ for file scope as per discord conversation

File: dDatabase/CDClientDatabase/CDClientTables/CDPhysicsComponentTable.cpp (hunk 6 above)
Comment (jadebenn) [reply dDatabase/CDClientDatabase/CDClientTables/CDPhysicsComponentTable.cpp:25]: I actually remember now: These used to be in an aggregate constructor. Guess I forgot to update the line endings.

**Comment by jadebenn** [comment]
Updated to account for feedback

**Comment by jadebenn** [comment]
Discovered an issue with this branch (#1493). Recommend holding off on review for now.

**Review by jadebenn — COMMENTED** [review]
Well. That was faster than I expected.

File: dDatabase/CDClientDatabase/CDClientTables/CDObjectsTable.cpp [hunk 7]
```diff
@@ -57,14 +56,16 @@ const CDObjects& CDObjectsTable::GetByID(uint32_t LOT) {
...
-		return m_default;
+		entries.emplace(LOT, ObjDefault);
+		return ObjDefault;
 	}
 
 	// Now get the data
 	while (!tableData.eof()) {
-		CDObjects entry;
-		entry.id = tableData.getIntField("id", -1);
+		const uint32_t LOT = tableData.getIntField("id", 0);
```
Comment (EmosewaMC) [inline dDatabase/CDClientDatabase/CDClientTables/CDObjectsTable.cpp:65]: LOT is a typedef, not sure how compilers handle a variable having the same name as a typedef

File: dDatabase/CDClientDatabase/CDClientTables/CDPhysicsComponentTable.cpp [hunk 8]
```diff
@@ -4,32 +4,31 @@ void CDPhysicsComponentTable::LoadValuesFromDatabase() {
...
-		entry.speed = tableData.getFloatField("speed", -1);
-		UNUSED(entry->rotSpeed = tableData.getFloatField("rotSpeed", -1));
-		entry.playerHeight = tableData.getFloatField("playerHeight");
-		entry.playerRadius = tableData.getFloatField("playerRadius");
+		UNUSED_COLUMN(entry.jump = tableData.getIntField("jump", -1) != 0;)
+		UNUSED_COLUMN(entry.doubleJump = tableData.getIntField("doublejump", -1) != 0;)
+		entry.speed = static_cast<float>(tableData.getFloatField("speed", -1));
+		UNUSED_COLUMN(entry.rotSpeed = tableData.getFloatField("rotSpeed", -1);)
+		entry.playerHeight = static_cast<float>(tableData.getFloatField("playerHeight"));
+		entry.playerRadius = static_cast<float>(tableData.getFloatField("playerRadius"));
```
Comment (EmosewaMC) [inline dDatabase/CDClientDatabase/CDClientTables/CDPhysicsComponentTable.cpp:18]: These are already floats in the entry struct, not sure why we're casting it to the type that is already returned here

**Review by EmosewaMC — APPROVED** [review]

File: dDatabase/CDClientDatabase/CDClientTables/CDPhysicsComponentTable.cpp (hunk 8 above)
Comment (EmosewaMC) [reply dDatabase/CDClientDatabase/CDClientTables/CDPhysicsComponentTable.cpp:18]: and that is of course because getFloatField does not return a float but a double.  Very cool api

File: dDatabase/CDClientDatabase/CDClientTables/CDObjectsTable.cpp (hunk 7 above)
Comment (jadebenn) [reply dDatabase/CDClientDatabase/CDClientTables/CDObjectsTable.cpp:65]: Seems the typedef gets shadowed, but I see your point.

**Review by aronwk-aaron — APPROVED** [review]

**Review by EmosewaMC — APPROVED** [review]
